File: app/src/main/java/com/sevtinge/cemiuiler/module/packageinstaller/DisableSafeModelTip.java
package com.sevtinge.cemiuiler.module.packageinstaller;

import com.sevtinge.cemiuiler.module.base.BaseHook;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.callbacks.XC_LoadPackage;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class DisableSafeModelTip extends BaseHook {
    @Override
    public void init() {

        findAndHookMethod("com.miui.packageInstaller.model.ApkInfo", "getSystemApp", XC_MethodReplacement.returnConstant(true));

        hookAllMethods("com.miui.packageInstaller.InstallProgressActivity", "g0", XC_MethodReplacement.returnConstant(false));

        XposedHelpers.findAndHookMethod("com.miui.packageInstaller.InstallProgressActivity", lpparam.classLoader, "Q1", new MethodHook() {
            @Override
            protected void before(MethodHookParam param) throws Throwable {
                param.setResult(new ArrayList<>());
            }
        });

        //returnIntConstant(findClassIfExists("p6.a"), "d");


    }


}

File: app/src/main/java/com/sevtinge/cemiuiler/module/packageinstaller/DisableSafeModelTip.kt
package com.sevtinge.cemiuiler.module.packageinstaller

import com.github.kyuubiran.ezxhelper.ClassUtils.loadClass
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHooks
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.findClassOrNull
import com.sevtinge.cemiuiler.utils.setBooleanField

object DisableSafeModelTip : BaseHook() {
    override fun init() {
        val miuiSettingsCompatClass =
            loadClass("com.android.packageinstaller.compat.MiuiSettingsCompat")

        try {
            miuiSettingsCompatClass.methodFinder().filterByName("isPersonalizedAdEnabled")
                .filterByReturnType(Boolean::class.java).toList().createHooks {
                before {
                    it.result = false
                }
            }
        } catch (t: Throwable) {
            logE(t)
        }

        var letter = 'a'
        for (i in 0..25) {
            try {
                val classIfExists =
                    "com.miui.packageInstaller.ui.listcomponets.${letter}0".findClassOrNull()
                classIfExists?.let {
                    it.methodFinder().filterByName("a").first().createHook {
                        after { hookParam ->
                            hookParam.thisObject.setBooleanField("l", false)
                        }
                    }
                }
            } catch (t: Throwable) {
                letter++
            }
        }
    }
}
